keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -84,8 +86,9 @@ public static Nd4jDoubleTensor arange(double start, double end, double stepSize)
 
     private INDArray tensor;
 
-    public Nd4jDoubleTensor(double[] data, int[] shape) {
-        this(TypedINDArrayFactory.create(data, shape, BUFFER_TYPE));
+    public Nd4jDoubleTensor(double[] data, long[] shape) {
+        TensorShapeValidation.checkRankIsAtLeastTwo(shape);
+        this.tensor = TypedINDArrayFactory.create(data, shape, BUFFER_TYPE);
```

Comment:
This looks like it needs merging from develop

Reply:
Good spot - bad merge on my part